applications/EigenSolversApplication/custom_solvers/feast_eigensystem_solver.h

```diff
@@ -0,0 +1,435 @@
+    class TDenseSpaceTypeIn = TUblasDenseSpace<TScalarIn>,
+    class TSparseSpaceTypeOut = TUblasSparseSpace<TScalarOut>,
+    class TDenseSpaceTypeOut = TUblasDenseSpace<TScalarOut>>
+class FEASTEigensystemSolver
+    : public LinearSolver<TSparseSpaceTypeIn, TDenseSpaceTypeIn>
+{
+    Parameters mParam;
+
+  public:
+    KRATOS_CLASS_POINTER_DEFINITION(FEASTEigensystemSolver);
+
+    typedef LinearSolver<TSparseSpaceTypeIn, TDenseSpaceTypeIn> BaseType;
+
+    typedef typename TSparseSpaceTypeIn::MatrixType SparseMatrixType;
+
+    typedef typename TSparseSpaceTypeIn::VectorType VectorType;
+
+    typedef typename TDenseSpaceTypeIn::MatrixType DenseMatrixType;
+
+    typedef typename TDenseSpaceTypeOut::MatrixType DenseOutputMatrixType;
+
+    typedef matrix<TScalarOut, column_major> FEASTMatrixType;
+
+    typedef typename TSparseSpaceTypeOut::VectorType OutputVectorType;
+
+    typedef TScalarIn ValueType;
+
+    FEASTEigensystemSolver(
+        Parameters param
+    ) : mParam(param)
+    {
+        Parameters default_params(R"(
+        {
+            "solver_type" : "eigen_feast",
+            "symmetric" : true,
+            "number_of_eigenvalues" : 0,
+            "search_lowest_eigenvalues" : false,
+            "search_highest_eigenvalues" : false,
+            "e_min" : 0.0,
+            "e_max" : 0.0,
+            "e_mid_re" : 0.0,
+            "e_mid_im" : 0.0,
+            "e_r" : 0.0,
+            "subspace_size" : 0,
+            "max_iteration" : 20,
+            "tolerance" : 1e-12,
+            "echo_level" : 0
+        })");
+
+        mParam.ValidateAndAssignDefaults(default_params);
+
+        KRATOS_ERROR_IF( mParam["number_of_eigenvalues"].GetInt() < 0 ) <<
+            "Invalid number of eigenvalues provided" << std::endl;
+
+        KRATOS_ERROR_IF( mParam["subspace_size"].GetInt() < 0 ) <<
+            "Invalid subspace size provided" << std::endl;
+
+        KRATOS_ERROR_IF( mParam["max_iteration"].GetInt() < 1 ) <<
+            "Invalid maximal number of iterations provided" << std::endl;
+
+        KRATOS_INFO_IF( "FEASTEigensystemSolver",
+            mParam["number_of_eigenvalues"].GetInt() > 0  && mParam["subspace_size"].GetInt() > 0 ) <<
+            "Manually defined subspace size will be overwritten to match the defined number of eigenvalues" << std::endl;
+
+        const TScalarOut T = {};
+        CheckParameters(T);
+    }
+
+    ~FEASTEigensystemSolver() override {}
+
+    /**
+     * Solve the generalized eigenvalue problem using FEAST
+     * @param rK first input matrix
+     * @param rM second input matrix
+     * @param rEigenvalues eigenvalues
+     * @param rEigenvectors row-aligned eigenvectors [n_evs,n_dofs]
+     */
+    void Solve(
+        SparseMatrixType& rK,
+        SparseMatrixType& rM,
+        VectorType& rEigenvalues,
+        DenseMatrixType& rEigenvectors) override
+    {
+        // settings
+        const size_t system_size = rK.size1();
```

Comment:
```suggestion
        const std::size_t system_size = rK.size1();
```

better to prescribe the namespace

Reply:
sure